soc: nxp: imxrt: Register log module to soc.c

asingh-GiN opened the pull request:
These changes fix a failing build on a IMXRT11xx, IMXRT5xx/CM33, and
IMXRT6xx/CM33 when debug logging is enabled. 

The error can be reproduced by running the following command: 
```
west build -p -b mimxrt1170_evk//cm7 samples/hello_world -DCONFIG_PM=y -DCONFIG_LOG=y -DCONFIG_LOG_DEFAULT_LEVEL=4
```

asingh-GiN committed:
soc: nxp: imxrt: Fix formatting in soc.c files

Fix formatting for `soc.c` files for the IMXRT11xx, IMXRT5xx/CM33, and
IMXRT6xx/CM33 by running 'clang-format'.

Signed-off-by: Apoorv Singh <apoorv.singh@gin.de>

decsny commented on soc/nxp/imxrt/imxrt11xx/soc.c:
```diff
@@ -15,6 +16,10 @@
 #include <fsl_dcdc.h>
+
+#define LOG_LEVEL CONFIG_SOC_LOG_LEVEL
+LOG_MODULE_REGISTER(soc);
```
nit, please do the log level as argument to the register macro

asingh-GiN committed:
soc: nxp:imxrt Register log module to soc.c

Previously, the `soc.c` files for the IMXRT11xx, IMXRT5xx/CM33, and
IMXRT6xx/CM33 did not register the log module. This caused build
errors when DEBUG logging was enabled, as the `power.c` file attempted
to access a non-existent SOC log module for debug messages.

This commit fixes the issue by registering the log module in the
`soc.c` files for the specified SoCs, thereby resolving the build
errors.

Signed-off-by: Apoorv Singh <apoorv.singh@gin.de>
